src/Public/Responses.ps1

```diff
@@ -785,6 +782,91 @@ function Write-PodeXmlResponse {
     Write-PodeTextResponse -Value $Value -ContentType 'text/xml' -StatusCode $StatusCode
 }
 
+
+
+
+<#
+.SYNOPSIS
+Writes YAML data to the Response.
+
+.DESCRIPTION
+Writes YAML data to the Response, setting the content type accordingly.
+
+.PARAMETER Value
+A String, PSObject, or HashTable value. For non-string values, they will be converted to YAML.
+
+.PARAMETER Path
+The path to a YAML file.
+
+.PARAMETER Depth
+The Depth to generate the YAML document - the larger this value the worse performance gets.
+
+.PARAMETER StatusCode
+The status code to set against the response.
+
+.EXAMPLE
+Write-PodeYamlResponse -Value '{"name": "Rick"}'
+
+.EXAMPLE
+Write-PodeYamlResponse -Value @{ Name = 'Rick' } -StatusCode 201
+
+.EXAMPLE
+Write-PodeYamlResponse -Path 'E:/Files/Names.json'
+#>
+function Write-PodeYamlResponse {
+    [CmdletBinding(DefaultParameterSetName = 'Value')]
+    param (
+        [Parameter(Mandatory = $true, ParameterSetName = 'Value')]
+        $Value,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'File')]
+        [string]
+        $Path,
+
+        [Parameter()]
+        [ValidatePattern('^\w+\/[\w\.\+-]+$')]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $ContentType = 'application/x-yaml',
+
+
+        [Parameter(ParameterSetName = 'Value')]
+        [validate]
+        [int]
+        $Depth = 10,
```

Comment:
`[validate]` is invalid and needs to be removed - unless it's a typo for ValidateRange or another Validate attribute? (JSON and the others don't have a validate on their Depth parameter)

Reply:
I wanted to add a range but I think I left this unfinished

Reply:
[ValidateRange(0,100)] was my goal